lib/timetable-form/Form.tsx

```diff
@@ -96,6 +101,27 @@ export const Form = (props: React.HTMLAttributes<HTMLDivElement>) => {
     reader.readAsText(file);
   }, []);
 
+  const handleValidateForm = useCallback(() => {
+    const validationErrors: ValidationErrorItem[] = [];
+
+    developmentPlanEvents.forEach((event) => {
```

Comment:
I am validating each stages individually here hence the loop. This is not really needed though as we could validate the whole array against an array shaped schema using Joi. I prefer this way as we can use the event's reference for the validation error path (see line 116). If we were to validate the array as a whole the resulting validation error's path would look like this [0, "eventDate"] (the first event in the events array failed validation for "eventDate") rather than ["event-unique-reference", "eventDate"]. Let me know if you have any thoughts on this